client/src/containers/widget/navigation/index.tsx

```diff
@@ -0,0 +1,72 @@
+import { FC } from "react";
```

Comment:
I would split this file into different two different components, FilterWidget and NavigationWidget, so there's no need to differentiate their data type and having conditional renders.